utils/submit_track_templates.py

```diff
@@ -129,59 +168,66 @@ def match_template(genome_id: str, datafile: str) -> None:
   if args.file and not args.template and datafile not in args.file:
     return
   filename = os.path.splitext(datafile)[0]
-  if f"{filename}.yaml" in template_files: # 1-to-1 match
+  # exact datafile=>template name match
+  if filename in templates:
     apply_template(genome_id, filename)
     return
-  # partial match (multiple tracks per datafile or vice versa)
-  match = False
-  for template_file in template_files:
-    if template_file.startswith(filename):
-      apply_template(genome_id, template_file)
-      match = True
-    elif not match and filename.startswith(os.path.splitext(template_file)[0]):
-      print(f"match to {template_file}")
-      apply_template(genome_id, template_file, datafile)
+  # partial name match (multiple tracks per datafile or vice versa)
+  multimatch = False
+  for template in templates:
+    # many templates per datafile (e.g. transcripts.bb)
+    if template.startswith(filename):
+      apply_template(genome_id, template)
+      multimatch = True
+    # one template for many datafiles (e.g. repeats.repeatmask*.bb)
+    if not multimatch and filename.startswith(template):
+      apply_template(genome_id, template, datafile)
       return
-  if not match:
-    if args.exclude:
-      for exclude in args.exclude:
-        if datafile.startswith(exclude):
-          return
-    print(f"Warning: No track template found for {datafile}")
+  # unexpected datafile name (fine for .bw since same track as .bb)
+  if not multimatch and datafile.endswith(".bb"):
+    log(f"Warning: No track template found for {datafile}")
 
 
 # 3) Fill in the template
-def apply_template(genome_id: str, template_file: str, datafile: str='') -> None:
-  if args.template and template_file not in args.template:
-    return
-  if(not template_file.endswith(".yaml")):
-    template_file += ".yaml"
-  with open(f"{template_dir}/{template_file}", "r") as file:
-    track_data: dict = yaml.safe_load(file)
+def apply_template(genome_id: str, template: str, datafile: str='') -> None:
+  with open(f"{template_dir}/{template}{EXT}", "r") as template_file:
+    track_data: TrackData = yaml.safe_load(template_file)
   track_data['genome_id'] = genome_id
-  if datafile: # update datafile (template matches multiple files)
+  # update datafile field (template matches multiple files)
+  if datafile:
     filename = os.path.splitext(datafile)[0]
     for key, value in track_data['datafiles'].items():
       if value.startswith(filename):
         track_data['datafiles'][key] = datafile
         break
-  # gene tracks need a species-specific description
-  if template_file.startswith("transcripts"):
-    if genome_id not in gene_desc:
-      log(f"Missing gene track description for genome {genome_id}. Skipping track.")
-      return
+  # fill in species-specific fields (gene & variation tracks)
+  if template.startswith("transcripts") or template.startswith("variant-ensembl"):
```

Comment:
Not that it is terribly important; but the variable name `template` is ambiguous. You are using it here as the name of a template file; but one might read is as referring to the contents of the template file (i.e. the template itself). `template_name` would have probably been clearer.

Reply:
Done